chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -409,21 +410,27 @@ func (b *BitcoindNotifier) handleRelevantTx(tx chain.RelevantTx, bestHeight int3
 func (b *BitcoindNotifier) historicalConfDetails(txid *chainhash.Hash,
 	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation, error) {
 
-	// First, we'll attempt to retrieve the transaction details using the
-	// backend node's transaction index.
+	// We'll attempt to retrieve the transaction details using the backend
+	// node's transaction index.
 	txConf, err := b.confDetailsFromTxIndex(txid)
 	if err != nil {
-		return nil, err
-	}
+		// If the transaction lookup was succesful, but it wasn't found
+		// within the index itself, then we'll exit early to avoid
+		// scanning the chain manually. We'll also need to look at the
+		// error message returned as the error code is used for multiple
+		// errors.
+		txNotFoundErr := "No such mempool or blockchain transaction"
```

Comment:
Have we confirmed that this is the exact help text? Ideally we also have a test along with this PR which will fail if the backend detects a query to the `GetBlocks` when we query for a transaction that was never even broadcast. The test would need to skip the neutrino backend, and only execute for the btcd and bitocind backends.